MH4U Dex/RegionItemTableViewController.m

```diff
@@ -0,0 +1,107 @@
+//
+//  RegionItemTableViewController.m
+//  MH4U Dex
+//
+//  Created by Joseph Goldberg on 3/6/15.
+//  Copyright (c) 2015 Joseph Goldberg. All rights reserved.
+//
+
+#import "RegionItemTableViewController.h"
+
+#import <CoreData/CoreData.h>
+
+#import "CoreDataController.h"
+
+#import "ItemContainerViewController.h"
+
+#import "Area.h"
+#import "AreaDrop.h"
+#import "Item.h"
+#import "Region.h"
+
+#import "AreaDropTableViewCell.h"
+
+@interface RegionItemTableViewController ()
+
+@property (nonatomic, strong) NSArray *areaArray;
+@property (nonatomic, strong) NSMutableDictionary *areaDropsDict;
+
+@end
+
+@implementation RegionItemTableViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    CoreDataController *coreDataController = [CoreDataController sharedCDController];
+    NSError *fetchError = nil;
+    NSSortDescriptor *areaSortDescriptor = [NSSortDescriptor sortDescriptorWithKey:AreaAttributes.name ascending:YES];
+    NSFetchRequest *areaFetchRequest = [[NSFetchRequest alloc] initWithEntityName:[Area entityName]];
+    NSPredicate *areaPredicate = [NSPredicate predicateWithFormat:@"%K == %@", AreaRelationships.region, self.region];
+    NSPredicate *rankPredicate;
+    if (self.rank) {
+        rankPredicate = [NSPredicate predicateWithFormat:@"%K == %@", AreaAttributes.rank, self.rank];
+    } else {
+        rankPredicate = [NSPredicate predicateWithFormat:@"%K == %@", AreaAttributes.rank, @"Low"];
+    }
+    NSCompoundPredicate *compoundPredicate = [NSCompoundPredicate andPredicateWithSubpredicates:@[rankPredicate, areaPredicate]];
+    [areaFetchRequest setPredicate:compoundPredicate];
+    [areaFetchRequest setSortDescriptors:@[areaSortDescriptor]];
+    self.areaArray = [coreDataController.managedObjectContext executeFetchRequest:areaFetchRequest error:&fetchError];
+    self.areaDropsDict = [NSMutableDictionary dictionary];
+    for (Area *area in self.areaArray) {
+        NSSortDescriptor *dropSortDescriptor = [NSSortDescriptor sortDescriptorWithKey:AreaDropAttributes.idDecimalString ascending:YES];
```

Comment:
Since this doesn't depend on the specific value of `area`, it would be more efficient to move it outside the loop.